### PR TITLE
Login storage refactor.

### PR DESCRIPTION
**Pull request by grigoryk**
Companion PR for https://github.com/mozilla-mobile/android-components/pull/6128

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/Core.kt [hunk 1]
```diff
@@ -201,6 +194,8 @@ class Core(private val context: Context) {
 
     val bookmarksStorage by lazy { PlacesBookmarksStorage(context) }
 
+    val passwordsStorage by lazy { PlacesLoginStorage(context, passwordsEncryptionKey) }
```
Comment (grigoryk) [inline app/src/main/java/org/mozilla/fenix/components/Core.kt]: We can now make `passwordsEncryptionKey` lazy!

**Comment by grigoryk** [comment]
Just waiting for https://github.com/mozilla-mobile/android-components/pull/6128 to be available in a snapshot.

**Comment by pocmo** [comment]
Snapshots are ready. Let's rerun this. :)

**Comment by pocmo** [comment]
This looks like it needs a rebase on top of master to address some other changes that have happened since opening the PR. I'll do that now.

**Comment by pocmo** [comment]
I still see some other failures locally:

```
e: /Users/sebastiankaspari/Projects/Mozilla/fenix/app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt: (332, 21): No value passed for parameter 'keyStore'
e: /Users/sebastiankaspari/Projects/Mozilla/fenix/app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt: (32, 47): Unresolved reference: SyncableLoginsStorage
e: /Users/sebastiankaspari/Projects/Mozilla/fenix/app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt: (53, 23): Unresolved reference: SyncableLoginsStorage
e: /Users/sebastiankaspari/Projects/Mozilla/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt: (40, 47): Unresolved reference: SyncableLoginsStorage
e: /Users/sebastiankaspari/Projects/Mozilla/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt: (76, 71): No value passed for parameter 'securePreferences'
e: /Users/sebastiankaspari/Projects/Mozilla/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt: (88, 71): No value passed for parameter 'securePreferences'
e: /Users/sebastiankaspari/Projects/Mozilla/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt: (195, 36): Unresolved reference: SyncableLoginsStorage
```

**Comment by pocmo** [comment]
Ah, I think this PR only contains the changes for the beta flavor. I'll adapt it and add the changes for the nightly flavor to it.

**Comment by pocmo** [comment]
Build passes, but tests not yet... looking. :)

**Comment by pocmo** [comment]
... and ktlint.

**Comment by pocmo** [comment]
Alright, green, phew. Lets get this in to unblock master.

**Pull request merged**